client/my-sites/design-menu/design-tools/header-image-control.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import get from 'lodash/get';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import ImagePreloader from 'components/image-preloader';
+import Button from 'components/button';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import EditorMediaModal from 'post-editor/media-modal';
+
+const HeaderImageControl = React.createClass( {
```

Comment:
For instance, this would be a `header-image` component outside of design menu, just a pure calypso component we can drop and use anywhere, since preview is decoupled.